enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,379 @@
+
+## Open Questions [optional]
+
+1. The bundles we build today are done via `docker` from the project's
+   `Makefile` where users can adapt their image builder to whatever they want.
+   Do we want to allow this from the CLI?
+   - if we do build images, we should add image builder flag
+1. What will happen to the existing command in SDK for deploying an operator
+   packagemanifests using OLM? Do we still want to support that and will it stay intact?
+   - we will deprecate the `run packagemanifests` command and remove in SDK 2.0
+
+## Summary
+
+Add a subcommand to the Operator SDK CLI that will allow users to migrate
+their operator's packagemanifest format to the new bundle format.
+
+## Motivation
+
+There are operators that exist that are still using the packagemanifest
+format. This subcommand will allow them to migrate these packagemanifests to
+bundles which is the preferred OLM packaging format.
+
+### Goals
+
+- Provide a subcommand to migrate one packagemanifest to a set of bundles.
+- Deprecate `packagemanifests` features:
+  - Deprecate `run packagemanifests`
+  - Deprecate `generate packagemanifests`
+
+### Non-Goals
+
+- Solve the world?
+- There will be no migration from bundles to packagemanifests
+
+## Proposal
+
+### User Stories [optional]
+
+#### Story 1
+
+As an operator developer, I would like to migrate my operator which uses
+the packagemanifest format to the new bundle format.
+
+#### Story 2
+
+As an catalog admin for my operators, I would like to be able to migrate a
+collection of packagemanifests to bundles.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports generating & validating
+packagemanifests & bundles. This enhancement will add a new command, `migrate
+bundle` to the CLI. This new command will migrate an existing packagemanifest to
+a bundle.
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+#### Generating bundle on disk
+
+In order to generate bundles to disk, we will refactor the `generate bundle`
+logic to be reusable in generating the bundles.
+
+===================== REWRITE BELOW =======================
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
```

Comment:
question; should we always need to inform the dir of the pkgmanifest? Cannot we assume that if was ot informed it ill be in the deploy/ dir since it was generated there before via sdk?

Reply:
@camilamacedo86 sorry I needed to rip this out :( but to answer your question. I don't think we should assume `deploy` because IIRC we should be able to do any packagemanifest not just operator-sdk generated ones.

Reply:
Would not we still able to do for any any packagemanifest? My suggestion here is : 

if packagemanifest-dir is not informed then set the default (deploy path) only that.

Reply:
@camilamacedo86 I guess we could, that would change the parameter to be a flag and not a required field. Right now the only parameter required is the packagemanifest directory:
```
operator-sdk pkgman-to-bundle <packagemanifestdir>
```
It just seems more explicit if I require the directory name. But if we can easily determine the default directory then maybe.